### ISSUE TITLE
`vhp find publicvulns` Fails with `undefined local variable or method` Error

### ISSUE THREAD

**Issue by nuthanmunaiah** [VulnerabilityHistoryProject/shepherd-tools#25]
# Description

`vhp find publicvulns` fails with `undefined local variable or method` error.

## Demonstration

```console
~/vhp/struts-vulnerabilities$ git remote -v
origin  https://github.com/vulnerabilityhistoryproject/struts-vulnerabilities (fetch)
origin  https://github.com/vulnerabilityhistoryproject/struts-vulnerabilities (push)

~/vhp/struts-vulnerabilities$ git status
HEAD detached at v3.0
nothing to commit, working tree clean

~/vhp/struts-vulnerabilities$ vhp --version
vhp 0.4.0

~/vhp/struts-vulnerabilities$ vhp find publicvulns --repo /scratch/repositories/struts/ --output struts.csv --start
2014-01-01 --end 2019-01-01
Getting vulnerable files list
Period start: 2014-01-01
Period end: 2019-01-01
Writing output file struts.csv
vhp 0.4.0 | Error:  undefined local variable or method `result' for #<VHP::VulnerableFileExtractor:0x000055efcad38c08>

~/vhp/struts-vulnerabilities$ cat struts.csv
filepath
```
### Environment

The environment in which `vhp` fails is as described in the output below.

```console
~/vhp/struts-vulnerabilities$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.5 LTS
Release:        18.04
Codename:       bionic

~/vhp/struts-vulnerabilities$ ruby --version
ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-linux-gnu]

~/vhp/struts-vulnerabilities$ rake --version
rake, version 13.0.3

~/vhp/struts-vulnerabilities$ gem list

*** LOCAL GEMS ***

bigdecimal (default: 1.3.4)
bundler (2.2.16)
byebug (11.1.3)
cmath (default: 1.0.0)
csv (default: 1.0.0)
date (default: 1.0.0)
dbm (default: 1.0.0)
did_you_mean (1.2.0)
diff-lcs (1.4.4)
etc (default: 1.0.0)
fcntl (default: 1.0.0)
fiddle (default: 1.0.0)
fileutils (default: 1.0.2)
gdbm (default: 2.0.0)
git (1.5.0)
httparty (0.17.0)
io-console (default: 0.4.6)
ipaddr (default: 1.2.0)
json (default: 2.1.0)
mercenary (0.3.6)
mime-types (3.3.1)
mime-types-data (3.2021.0225)
minitest (5.10.3)
multi_xml (0.6.0)
net-telnet (0.1.1)
openssl (default: 2.1.1)
os (1.1.0)
parallel (1.13.0)
power_assert (0.2.7)
psych (default: 3.0.2)
rake (13.0.3, 12.3.1)
rdoc (default: 6.0.1)
require_all (3.0.0)
rspec (3.10.0)
rspec-core (3.10.1)
rspec-expectations (3.10.1)
rspec-mocks (3.10.2)
rspec-support (3.10.2)
ruby-progressbar (1.10.1)
scanf (default: 1.0.0)
sdbm (default: 1.0.0)
stringio (default: 0.0.1)
strscan (default: 1.0.0)
test-unit (3.2.5)
vhp (0.4.0)
webrick (default: 1.4.2)
zlib (default: 1.0.0)
```

**Comment by Anorwen** [comment]
I'm taking a look at this issue: it looks like there was a line commented out because it was outdated. I'll work on updating this place in the code.

Found on line 41 of vuln-files.rb
```
      # FIXME Update to new GitAPI
      # result = GitLog.new(@options[:repo]).get_files_from_shas(fixes)
```

**Comment by Anorwen** [comment]
I just created pr #26 which should fix this issue. Can you pull from branch `issue-25-result`, install it, and give the command another try. I believe it worked for me but I've not as familiar with the tool.

**Comment by nuthanmunaiah** [comment]
The issue seems to be resolved in version 0.5.0a of VHP.

```console
~/vhp/struts-vulnerabilities$ vhp --version
vhp 0.5.0a

~/vhp/struts-vulnerabilities$ vhp find publicvulns --repo /scratch/repositories/struts/ --output struts.csv --start
2014-01-01 --end 2019-01-01
Getting vulnerable files list
Period start: 2014-01-01
Period end: 2019-01-01
Writing output file struts.csv

~/vhp/struts-vulnerabilities$ wc -l struts.csv
537 struts.csv
```
